pkg/evpn/svi.go

```diff
@@ -27,88 +27,122 @@ import (
+		resourceID = in.SviId
 	}
-	in.Tunnel.Name = resourceIDToFullName("tunnels", resourceID)
+	in.Svi.Name = resourceIDToFullName("svis", resourceID)
 	// idempotent API when called with same key, should return same object
-	obj, ok := s.Tunnels[in.Tunnel.Name]
+	obj, ok := s.Svis[in.Svi.Name]
 	if ok {
-		log.Printf("Already existing Tunnel with id %v", in.Tunnel.Name)
+		log.Printf("Already existing Svi with id %v", in.Svi.Name)
 		return obj, nil
 	}
 	// not found, so create a new one
-	myip := make(net.IP, 4)
-	binary.BigEndian.PutUint32(myip, in.Tunnel.Spec.LocalIp.GetV4Addr())
-	vxlanid := in.Tunnel.Spec.Encap.Value.GetVnid()
-	// TODO: take Port from proto instead of hard-coded
-	vxlan := &netlink.Vxlan{LinkAttrs: netlink.LinkAttrs{Name: resourceID}, VxlanId: int(vxlanid), Port: 4789, Learning: false, SrcAddr: myip}
-	if err := netlink.LinkAdd(vxlan); err != nil {
-		fmt.Printf("Failed to create link: %v", err)
-		return nil, err
-	}
-	// TODO: gap... instead of VPC we are looking for Subnet/Bridge here...
-	if in.Tunnel.Spec.VpcNameRef != "" {
-		// Validate that a Subnet/Bridge resource name conforms to the restrictions outlined in AIP-122.
-		if err := resourcename.Validate(in.Tunnel.Spec.VpcNameRef); err != nil {
-			log.Printf("error: %v", err)
-			return nil, err
-		}
-		// now get Subnet/Bridge to plug this vxlan into
-		bridge, ok := s.Bridges[in.Tunnel.Spec.VpcNameRef]
-		if !ok {
-			err := status.Errorf(codes.NotFound, "unable to find key %s", in.Tunnel.Spec.VpcNameRef)
-			log.Printf("error: %v", err)
-			return nil, err
-		}
-		brdev, err := netlink.LinkByName(path.Base(bridge.Name))
-		if err != nil {
-			err := status.Errorf(codes.NotFound, "unable to find key %s", bridge.Name)
-			log.Printf("error: %v", err)
+	bridge, err := netlink.LinkByName(tenantbridgeName)
+	if err != nil {
+		err := status.Errorf(codes.NotFound, "unable to find key %s", tenantbridgeName)
+		log.Printf("error: %v", err)
+		return nil, err
+	}
+	// Validate that a LogicalBridge resource name conforms to the restrictions outlined in AIP-122.
+	if err := resourcename.Validate(in.Svi.Spec.LogicalBridge); err != nil {
+		log.Printf("error: %v", err)
+		return nil, err
+	}
+	// now get LogicalBridge object to fetch VID field
+	bridgeObject, ok := s.Bridges[in.Svi.Spec.LogicalBridge]
+	if !ok {
+		err := status.Errorf(codes.NotFound, "unable to find key %s", in.Svi.Spec.LogicalBridge)
+		log.Printf("error: %v", err)
+		return nil, err
+	}
+	vid := uint16(bridgeObject.Spec.VlanId)
+	// Example: bridge vlan add dev br-tenant vid <vlan-id> self
+	if err := netlink.BridgeVlanAdd(bridge, vid, false, false, true, false); err != nil {
+		fmt.Printf("Failed to add vlan to bridge: %v", err)
+		return nil, err
+	}
+	// Example: ip link add link br-tenant name <link_svi> type vlan id <vlan-id>
+	vlandev := &netlink.Vlan{
+		LinkAttrs: netlink.LinkAttrs{
+			Name:        resourceID,
+			ParentIndex: bridge.Attrs().Index,
+		},
+		VlanId: int(bridgeObject.Spec.VlanId),
```

Comment:
How this code snippet is equivalent to the 
ip link add link br-tenant name <link_svi> type vlan id <vlan-id> 
command I do not see the realtion. From what I understnad you just create a netlink.Vlan object but you do not execute any netlink function that can justify the command in the example. no ?

Reply:
You are absolutely correct, thanks for catching this. My bad. Was in the hurry and missed it